components/script/dom/fakexrdevice.rs

```diff
@@ -66,45 +62,52 @@ pub fn get_views(views: &[FakeXRViewInit]) -> Fallible<(MockVRView, MockVRView)>
     let mut proj_r = [0.; 16];
     let v: Vec<_> = left.projectionMatrix.iter().map(|x| **x).collect();
     proj_l.copy_from_slice(&v);
+    let proj_l = TypedTransform3D::from_array(proj_l);
     let v: Vec<_> = right.projectionMatrix.iter().map(|x| **x).collect();
     proj_r.copy_from_slice(&v);
+    let proj_r = TypedTransform3D::from_array(proj_r);
+
+    let offset_l = get_origin(&left.viewOffset)?;
+    let offset_r = get_origin(&left.viewOffset)?;
 
-    let mut offset_l = [0.; 3];
-    let mut offset_r = [0.; 3];
-    let v: Vec<_> = left.viewOffset.position.iter().map(|x| **x).collect();
-    offset_l.copy_from_slice(&v);
-    let v: Vec<_> = right.viewOffset.position.iter().map(|x| **x).collect();
-    offset_r.copy_from_slice(&v);
-    let left = MockVRView {
+    let left = View {
         projection: proj_l,
-        offset: offset_l,
+        transform: offset_l,
     };
-    let right = MockVRView {
+    let right = View {
         projection: proj_r,
-        offset: offset_r,
+        transform: offset_r,
     };
-    Ok((left, right))
+    Ok(Views::Stereo(left, right))
 }
 
-pub fn get_origin(origin: &FakeXRRigidTransformInit) -> Fallible<([f32; 3], [f32; 4])> {
+pub fn get_origin<T, U>(
+    origin: &FakeXRRigidTransformInit,
+) -> Fallible<TypedRigidTransform3D<f32, T, U>> {
     if origin.position.len() != 3 || origin.orientation.len() != 4 {
         return Err(Error::Type("Incorrectly sized array".into()));
     }
-    let mut p = [0.; 3];
-    let mut o = [0.; 4];
-    let v: Vec<_> = origin.position.iter().map(|x| **x).collect();
-    p.copy_from_slice(&v[0..3]);
-    let v: Vec<_> = origin.orientation.iter().map(|x| **x).collect();
-    o.copy_from_slice(&v);
+    let p = TypedVector3D::new(
+        *origin.position[0],
+        *origin.position[1],
+        *origin.position[2],
+    );
+    let o = TypedRotation3D::unit_quaternion(
+        *origin.orientation[0],
+        *origin.orientation[1],
+        *origin.orientation[2],
+        *origin.orientation[3],
+    );
```

Comment:
Can we store a `TypedRotation3D` in `FakeXRTransformInit`?

Reply:
No, it's a dictionary, we don't control it.

Reply:
Oh well.